renderdoc/driver/vulkan/vk_core.cpp

```diff
@@ -705,6 +705,78 @@ void WrappedVulkan::SetDebugMessageSink(WrappedVulkan::ScopedDebugMessageSink *s
   Threading::SetTLSValue(debugMessageSinkTLSSlot, (void *)sink);
 }
 
+void WrappedVulkan::InsertPendingCommandBufferCallbacksEvent(VkCommandBuffer commandBuffer)
+{
+  // This occurs pre-baking as the event needs to be in the command buffer before vkEndCommandBuffer
+  // is called
+
+  VkResourceRecord *cmdRecord = GetRecord(commandBuffer);
+  VkPendingSubmissionCompleteCallbacks *pending =
+      cmdRecord->cmdInfo->pendingSubmissionCompleteCallbacks;
+  RDCASSERT(pending->event == VK_NULL_HANDLE);
+
+  if(pending->callbacks.empty())
+    return;
+
+  rdcarray<std::function<void()>> callbacks;
+  callbacks.swap(pending->callbacks);
```

Comment:
I'm not sure I understand the reason for swapping these callbacks into a local array only to move them back to the pending array at the end of this function?